### PR TITLE
First review

### PR DESCRIPTION
**Pull request by DavidBraslow**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,11 @@
...
+## Setup & Installtion
+
+Make sure you have the latest version of Python installed.
+
+    git clone <repo-url>
+    pip install -r requirements.txt
+  
+## Running The App
+
+    python app.py
```
Comment (jennettageorge) [inline README.md:11]: > you will need to do more than just run app.py to get the app to work. Notice that in app.py, the `predict` endpoint has the following line of code:

    ` raw_tweet = request.form.values()[0]`

    > what this tells us os that you are submitting a form and taking the first entry as the raw tweet to be ran against your model. So you will need to update your readme to relect the actual steps that the user needs to take to get the app to run successfully

**Comment by jennettageorge** [comment]
In app.py you have the line

`prediction = model.predict(final_features)`

 but `final_features` is not defined - can you fix this so we know what should actually go here?

**Comment by jennettageorge** [comment]
In the line above that, you have the line of code

`vectorized_tweet = `

this will throw a syntax error, since nothing is on the right hand side of the equal sign. Did you mean to have something here?

**Comment by jennettageorge** [comment]
Line 45 of pre_processing.py you have

`stops = stops `

this line of code doesn't do anything, did you mean to do something else here?

**Comment by jennettageorge** [comment]
are you sure you need all of those requirements in your `requirement.txt` file? also it throws syntax errors because of your `=` instead of `==` syntax, this will need to be fixed to compile

(hint: try starting out with a super easy file that only has this in it:

```
numpy
flask
pickle-mixin
pandas
nltk
sklearn
```

**Comment by jennettageorge** [comment]
you're missing a very important file. We will need to create a file called `request.py` that handles the post request for the app. The functionality of the app will be to receive and process the incoming tweet and feed it to the app.

Just to give you an idea of something I threw together super fast for the sake of error handling:

```
import requests

url = 'http://localhost:5000/results'

r = requests.post(url,json={'tweet':'This is a test tweet yay!'})

if r:
    print(r.json())
else:
    print("Nothing to return")
```

And this is the error we get when we run this:

```
Traceback (most recent call last):
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 2088, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/jennetta/Projects/twitter_sentiment/app.py", line 37, in results
    prediction = model.predict([np.array(list(data.values()))])
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/sklearn/naive_bayes.py", line 74, in predict
    X = self._check_X(X)
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/sklearn/naive_bayes.py", line 474, in _check_X
    return check_array(X, accept_sparse='csr')
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/sklearn/utils/validation.py", line 63, in inner_f
    return f(*args, **kwargs)
  File "/Users/jennetta/Projects/twitter_sentiment/env/lib/python3.7/site-packages/sklearn/utils/validation.py", line 714, in check_array
    "into decimal numbers with dtype='numeric'") from e
ValueError: Unable to convert array of bytes/strings into decimal numbers with dtype='numeric'
```

this will be the last comment I leave until you have some time to work on this ... moral of the story is we are still a good deal away from having a running app, but im here to help! Let me know how I can assist you in making this functional.

**Review by jennettageorge — CHANGES_REQUESTED** [review]

**Comment by DavidBraslow** [comment]
> are you sure you need all of those requirements in your `requirement.txt` file? also it throws syntax errors because of your `=` instead of `==` syntax, this will need to be fixed to compile
> 
> (hint: try starting out with a super easy file that only has this in it:
> 
> ```
> numpy
> flask
> pickle-mixin
> pandas
> nltk
> sklearn
> ```

Thanks for the tip. This was the automatic output from the conda environment I've been working in. Would you recommend I use virtualenv instead, or something else, to better emulate the environment & setup for AWS?

**Pull request merged**